### ISSUE TITLE
Impossible to purge the DB

### ISSUE THREAD

**Issue by redge76** [tananaev/traccar-client-android#151]
Hi,
I found some annoying "feature". I don't know if the solution is in the server of the client: But:
Because the android client store the positions in a DB and tries to send them in FIFO and it does not consider then as sent if the server does not reply, if you run the traccar client with a certain device ID, without adding the client ID on the server (so the DB records the positions) and then change the ID of the client to something else and then add the ID on the server, the new positions never reach the server because the client is unable to send the old position with the old id.
Solution could be:
- truncate the DB when the ID change
- Make sure that the server returns something even if the ID is unknown. (currently the RequestManager.sendRequest() thown an IOException if the ID in unknown on the server.
Because the catch only retrun false without any Log.e() , it was not obvious.
- use the HTTP error code returned by the server. Currently if there is a problem sending the position to the server, the user is not warned.

(Too complicated for me to supply a pull request...)

Redge

**Comment by tananaev** [comment]
First of all, can you please explain the use case for ID change.

I think if the ID is unknown on the server, it makes sense to return an error.

If there is a problem with sending data, you should be able to see it on the status screen. Can you please check and if it's not there report a bug.

**Comment by redge76** [comment]
Well I don't think there is a real "use case" in changing the ID. But is is available in the client interface.
My "stupid" use case is because the ID is too long to type and I don't want to go check in the traccar server log for the ID to do a copy/paste, I just change it to "a". Then I use this ID in the web interface. :-) Yes I'm lazy
So another solution would be to disable ID change if there is no point in letting the user do it.
My main concern is that I got stuck by changing the id and nothing was displayed to help me to diagnose this. I had to look at the log and recognize that the sent ID was the old ID. 
Then I had to add a device with the old ID to purge the DB and then it worked.

Everything works as designed. That why I called this a "feature". But it's counter intuitive,( I think)

Another solution would be to add the current device ID to the query in the ProtocolFormatter.java and not use the one stored in DB. Why store the device ID  for each position in the DB ?

> If there is a problem with sending data, you should be able to see it on the status screen
No, nothing is displayed. (and there is nothing in the code to display something. There is only 3 StatusActivity.addMessage() in the code and nothing to display an error)
Anyhow it's not really an error, just the server won't accept the position update until a correct ID is added to the interface.

How do I fill a bug rapport?
